src/App.js

```diff
@@ -138,36 +169,123 @@ function App() {
-                  <WiredLink
-                    href={`https://www.google.com/maps/search/?api=1&query=${pickedMerchant.location}`}
-                    target="_blank"
-                    rel="noopener"
-                    className="txt-cta"
-                  >
-                    Open in Map
-                  </WiredLink>
-                  <br />
-                  <WiredLink
-                    href={pickedMerchant.link}
-                    target="_blank"
-                    rel="noopener"
-                    className="txt-cta"
+                  <p className="txt-resto text-center">{pickedMerchant.name}</p>
+
+                  <div className="text-center spacing-x-8">
+                    <WiredLink
+                      href={`https://www.google.com/maps/search/?api=1&query=${pickedMerchant.location}`}
+                      target="_blank"
+                      rel="noopener"
+                      className="txt-cta"
+                    >
+                      Open in Map
+                    </WiredLink>
+
+                    <span>&bull;</span>
+
+                    <WiredLink
+                      href={pickedMerchant.link}
+                      target="_blank"
+                      rel="noopener"
+                      className="txt-cta"
+                    >
+                      Open in GoFood
+                    </WiredLink>
+                  </div>
+
+                  <WiredButton
+                    elevation={2}
+                    onClick={() => setIsResultModalOpen(true)}
                   >
-                    Open in GoFood
-                  </WiredLink>
-                  {/* <span>--MENU--</span>
-              {pickedMenus.map((menu, key) => (
-                <div key={key}>
-                  <p>{menu.name}</p>
-                  <img alt={menu.name} src={menu.image} />
-                  <p>{menu.price}</p>
-                </div>
-              ))} */}
+                    See restaurant detail
+                  </WiredButton>
                 </>
               )}
+
               <p className="foot-notes">&copy; @ans4175</p>
             </section>
+
+            <WiredDialog open={isResultModalOpen}>
+              {fetched && (
+                <section className="result-modal-section">
+                  <div className="result-modal-resto-information">
+                    <div>
+                      <h2 className="txt-resto text-center">
+                        {pickedMerchant.name}
+                      </h2>
+
+                      {/* TODO(imballinst): instead of bull-separated, we can add SVG icons instead. */}
+                      <ul className="list-style-none divider-bull flex txt-resto-info">
+                        {pickedMerchant.eta_cooking_minutes ||
+                        pickedMerchant.eta_delivery_minutes ? (
+                          <li>
+                            {pickedMerchant.eta_cooking_minutes +
```

Comment:
what if one got null/undefined, math additional will not error?

Reply:
...bener juga, saya terlalu berasumsi. Siap, nanti saya bikin default jadi 0 kalau `null` atau `undeifned`, sehingga ngga jadi `NaN`